### ISSUE TITLE
[bug] JIT, webpack 4 lazyloaded module: Unhandled Promise rejection: Can't resolve all parameters for

### ISSUE THREAD

**Issue by KrakenTyio** [rangle/augury#1306]
Augury version (required): 1.16.2
Angular version (required): 5-6
Date: 17.5.18
OS: arch gnome

Description of issue:
error described here
[angular issues](https://github.com/angular/angular/issues/23958)

**Comment by cgatian** [comment]
Also seeing this issue

**Comment by santiago-elustondo** [comment]
Thanks for reporting. This issue is top priority.

**Comment by LuisPalomo** [comment]
Same problem here when injecting a service into a lazy load module.
`ERROR Error: Uncaught (in promise): Error: Can't resolve all parameters for TranslateService: (?, ?, ?, ?, ?, [object Object], [object Object]).`

**Comment by santiago-elustondo** [comment]
Thanks for the clue, @LuisPalomo.
Can you tell me more about the service you're injecting?
- How are you injecting the service? (as a constructor param, with the `@Inject()` decorator, with `this.injector`, etc)
- What module is providing the service? Is it provided by the lazy loaded module itself?
- What are the service's own dependencies? What modules do they come from? Are they also lazy-loaded?

**Comment by LuisPalomo** [comment]
I'm using:
```
Angular CLI: 1.6.6
Node: 8.4.0
OS: win32 x64
Angular: 5.2.2
```

The service is the `TranslateService` from `@ngx-translate/core": "^9.1.1`
It's injected in the constructor of the root component of the lazy loaded module:
```
@NgModule({
    imports: [
        ...
        TranslateModule.forChild({
            loader: {
                provide: TranslateLoader,
                useFactory: HttpLoaderFactory,
                deps: [
                  HttpClient,
                ],
            },
        }),
    ],
    providers: [
        ...
    ],
    declarations: [
        PhysicalPersonModificationComponent,
        ...
    ],
})
export class PhysicalPersonModificationModule {}
```
```
@Component({...})
export class PhysicalPersonModificationComponent {
    constructor(private translateService: TranslateService) {...}
}
```

The module that is providing the service is the `TranslateModule` that is imported in the core module of the app:
```
@NgModule({
    imports: [
        ...
        TranslateModule.forRoot({
            loader: {
                provide: TranslateLoader,
                useFactory: HttpLoaderFactory,
                deps: [
                  HttpClient,
                ],
            },
        }),
    ],
    providers: [
        ...
    ],
    declarations: [
         ...
    ],
})
export class CoreModule {}
```
Hope it helps! 👍

**Comment by santiago-elustondo** [comment]
Thanks, @LuisPalomo.

Issue replicated with the following conditions:
Lazy loaded module provides a service that has not been previously provided by a parent module, and that requires dependency resolutions.

**Comment by staeke** [comment]
The issue appears on e.g. Webpack’s hot module reload - maybe especially if you’ve loaded your whole angular app and reflect-metadata polyfill lazily. And as I said - somehow Augury’s internal version of reflect-metadata (backend.js) takes precedence. Thus two different versions of Reflect.metadata APIs are used at consumption time and definition time

**Comment by andrewthauer** [comment]
We've published a rolled back version (1.16.3) to address this issue, since it seems to affect multiple people. Unfortunately we were not able to reproduce this issue (in a variety of applications). That said, it is likely related to `reflect-metadata` as pointed out and we have a working solution to avoid this issue moving forward (likely a Canary build tomorrow).

**Comment by vgordievskiy** [comment]
I have had this problem after I have updated the Chrome to 66 version

**Comment by santiago-elustondo** [comment]
A new version of the [Canary Build](https://chrome.google.com/webstore/detail/augury-canary-build/leechohfifidanmkioncamdcibhmmcji) (1.17.0) is released, with what we believe might be a fix for this issue. 

reflect-metadata has been removed from the bundle. Angular uses reflect-metadata for its JIT compilations, and a versioning conflict was occurring for some users. We now piggyback off of Angular's global reflect-metadata polyfill.

Can you confirm / disconfirm that version 1.17.0 of the [Canary Build](https://chrome.google.com/webstore/detail/augury-canary-build/leechohfifidanmkioncamdcibhmmcji) fixes the issue?

**Comment by santiago-elustondo** [comment]
A new version of the [Canary Build](https://chrome.google.com/webstore/detail/augury-canary-build/leechohfifidanmkioncamdcibhmmcji) (1.17.0) is released, with what we believe might be a fix for this issue. 

reflect-metadata has been removed from the bundle. Angular uses reflect-metadata for its JIT compilations, and a versioning conflict was occurring for some users. We now piggyback off of Angular's global reflect-metadata polyfill.

@elboukhari , @williamchristophercooley , @s001dxp , @jaj1014 , @mxchange , can you confirm / disconfirm that version 1.17.0 of the [Canary Build](https://chrome.google.com/webstore/detail/augury-canary-build/leechohfifidanmkioncamdcibhmmcji) fixes the issue?

**Comment by s001dxp** [comment]
@santiago-elustondo, The 1.17.0 build works. I have found, however, that the state is only showing for some components and not others...but that a separate bug and I'll post in a new issue if there is not already one created for it.

Thanks!

**Comment by santiago-elustondo** [comment]
Canary merged with master as official 1.18 release.
